src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1696,8 +1706,41 @@ private void NotifyValueChanged(TransactionOperationContext context, string type
             };
         }
 
+        [Flags]
+        public enum DatabaseRecordChange
+        {
+            None = 0,
+            Index = 1,
+            Backup = 1 << 1,
+            Replication = 1 << 2,
+            Etl = 1 << 3,
+            Subscription = 1 << 4,
+            Configuration = 1 << 5,
+
+            All = Index |
```

Comment:
We need test / debug assert to verify that we aren't adding a new type here and not updating the `All`